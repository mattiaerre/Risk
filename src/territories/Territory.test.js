const australia = require('../data/australia.json');
const Territory = require('./Territory');

test('Eastern Australia', () => {
  const easternAustralia = new Territory(australia[0]);

  expect(easternAustralia).toMatchSnapshot();

  const newGuinea = new Territory(australia[2]);

  easternAustralia.setBorders([newGuinea]);

  expect(easternAustralia.bordersWith('indonesia')).toBe(false);

  expect(easternAustralia.bordersWith('new-guinea')).toBe(true);
});
