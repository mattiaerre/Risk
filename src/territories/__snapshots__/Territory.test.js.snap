// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Eastern Australia 1`] = `
Territory {
  "borders": Array [],
  "id": "eastern-australia",
  "name": "Eastern Australia",
}
`;
